aws/src/test/java/org/apache/iceberg/aws/s3/TestS3URI.java

```diff
@@ -90,4 +92,17 @@ public void testValidSchemes() {
       assertEquals("path/to/file", uri.key());
     }
   }
+
+  @Test
+  public void testS3URIWithBucketToAccessPointMapping() {
```

Comment:
might be a bit hard, but is it possible to run with integration test by creating an access point on the fly?

Reply:
I am thinking of finding a way to do that, As per my understanding this would require creating a `S3ControlClient` as per https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/s3control/S3ControlClient.html#createAccessPoint-- which available at sdk version >= `2.17.149` tried searching it my intellij maybe at present version (`2.17.131`) it's available with diff name if present. S3MockRule at present doesn't have a way to test this atleast from what I can say which is what is used in the `Tests3OutputStream`

once the above is sorted I believe it rest of stuff should be easy to follow.